amundsen_application/static/js/components/common/ResourceList/PaginatedApiResourceList/index.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import Pagination from 'react-js-pagination';
+import ResourceListItem from 'components/common/ResourceListItem';
+import { Resource } from 'interfaces';
+import * as Constants from '../constants';
+
+import '../styles.scss';
+
+export interface PaginatedApiResourceListProps {
+  activePage: number;
+  emptyText?: string;
+  itemsPerPage: number;
+  onPagination: (pageNumber: number) => void;
+  slicedItems: Resource[];
+  totalItemsCount: number;
+  source: string;
+}
+
+class PaginatedApiResourceList extends React.Component<PaginatedApiResourceListProps, {}> {
+  public static defaultProps: Partial<PaginatedApiResourceListProps> = {
+    emptyText: Constants.DEFAULT_EMPTY_TEXT,
+  };
+
+  constructor(props) {
+    super(props);
+  }
+
+  onPagination = (rawPageNum: number) => {
+    const activePage = rawPageNum - 1;
+    this.props.onPagination(activePage);
+  };
+
+  render() {
+    const { activePage, emptyText, totalItemsCount, itemsPerPage, slicedItems, source } = this.props;
+    const startIndex = itemsPerPage * activePage;
+
+    return (
+      <div className="resource-list">
+        {
+          totalItemsCount === 0 && emptyText &&
+          <div className="empty-message body-placeholder">
+            { emptyText }
+          </div>
+        }
+        {
+          totalItemsCount > 0 &&
+          <>
+            <ul className="list-group">
+              {
+                slicedItems.map((item, idx) => {
+                  const logging = { source, index: startIndex + idx };
+                  return <ResourceListItem item={ item } logging={ logging } key={ idx } />;
+                })
+              }
+            </ul>
+            {
+              totalItemsCount > itemsPerPage &&
+              <Pagination
+                activePage={ activePage + 1 }
+                itemsCountPerPage={ itemsPerPage }
+                totalItemsCount={ totalItemsCount }
+                pageRangeDisplayed={ Constants.PAGINATION_PAGE_RANGE }
+                onChange={ this.onPagination }
+              />
+            }
+          </>
+        }
+        <div className="resource-list-footer" />
```

Comment:
Do we need this div?

Reply:
Nope. I  will also have to update some styles if I remove it to maintain the space we need at the bottom of the lists.